blockindex/contractstaking/delta_state.go

```diff
@@ -27,6 +27,7 @@ var (
 		},
 		deltaStateAdded: {
 			deltaActionModify: deltaStateAdded,
+			deltaActionRemove: deltaStateUnchanged,
```

Comment:
what about `deltaStateRemoved`? does it need a state-transition?

Reply:
also need to check `blkHeight == currentHeight + 1` in `PutBlock()`?

Reply:
> what about `deltaStateRemoved`? does it need a state-transition?

no valid action could be triggered on `deltaStateRemoved`

Reply:
> as we chatted, also need to check `blkHeight == currentHeight + 1` in `PutBlock()`?

Thx for reminding

Reply:
add a test to validate, once a bucket is deleted, i.e., `deltaStateRemoved`, it cannot transition to any further state (in the same block)